components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AllergiesController.java

```diff
@@ -0,0 +1,198 @@
+
+    @Override
+    protected String getJsonPropertyName()
+    {
+        return getName();
+    }
+
+    @Override
+    protected List<String> getProperties()
+    {
+        return Arrays.asList(ALLERGIES, NKDA);
+    }
+
+    @Override
+    protected List<String> getBooleanFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected List<String> getCodeFields()
+    {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public PatientData<Object> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+            BaseObject data = doc.getXObject(Patient.CLASS_REFERENCE);
+            if (data == null) {
+                throw new NullPointerException("The patient does not have a PatientClass");
+            }
+
+            Map<String, Object> result = new LinkedHashMap<String, Object>();
+
+            // allergies
+            ListProperty allergiesListProperty = (ListProperty) data.get(ALLERGIES);
+            List<String> allergiesList = allergiesListProperty.getList();
+            result.put(ALLERGIES, allergiesList);
+
+            // NKDA
+            IntegerProperty nkdaProperty = (IntegerProperty) data.get(NKDA);
+            Integer nkdaInteger = (Integer) nkdaProperty.getValue();
+            Boolean nkda = Boolean.TRUE;
+            if (nkdaInteger != 1) {
+                nkda = Boolean.FALSE;
+            }
+            result.put(NKDA, nkda);
+
+            return new DictionaryPatientData<>(getName(), result);
+        } catch (Exception e) {
+            this.logger.error(
+                "Could not find requested document or some unforeseen error has occurred during controller loading");
+        }
+        return null;
+    }
+
+    @Override
+    public void writeJSON(Patient patient, JSONObject json, Collection<String> selectedFieldNames)
+    {
+        if (selectedFieldNames != null && !selectedFieldNames.contains(ALLERGIES)) {
+            return;
+        }
+
+        PatientData<Object> allergiesData = patient.getData(DATA_NAME);
+        if (allergiesData == null) {
+            return;
+        }
+
+        @SuppressWarnings("unchecked")
+        List<String> allergies = (List<String>) allergiesData.get(ALLERGIES);
+        Boolean nkda = (Boolean) allergiesData.get(NKDA);
+
+        if (nkda != null && nkda.booleanValue()) {
+            json.put(ALLERGIES, NKDA);
+            return;
+        }
+
+        if (allergies != null && allergies.size() > 0) {
+            JSONArray allergiesArray = new JSONArray();
+            allergiesArray.addAll(allergies);
+            json.put(ALLERGIES, allergiesArray);
+        }
+    }
+
+    @Override
+    public PatientData<Object> readJSON(JSONObject json)
+    {
+        Map<String, Object> result = new LinkedHashMap<>();
+        List<String> allergiesList = new LinkedList<>();
+        Boolean nkda = Boolean.FALSE;
```

Comment:
You don't have to keep using `Boolean`, you can just use the primitive type `boolean`, and java will autobox it to `Boolean` whenever a function requires it.